ext/pdo_firebird/php_pdo_firebird_int.h

```diff
@@ -59,6 +59,11 @@ typedef void (*info_func_t)(char*);
 # define PDO_FIREBIRD_HANDLE_INITIALIZER NULL
 #endif
 
+typedef struct {
+	int sqlcode;
+	char *errmsg;
+} pdo_firebird_error_info;
```

Comment:
Maybe add the length component so that we don't need to recompute known lengths when converting this to zend_string ?
```suggestion
typedef struct {
	int sqlcode;
	char *errmsg;
	size_t errmsg_length;
} pdo_firebird_error_info;
```

Reply:
That's a good idea. Changed `pdo_firebird_fetch_error_func` to use `add_next_index_stringl`.